src/lib/utils/dissovled_oxygen_compensation.h

```diff
@@ -0,0 +1,94 @@
+#include <cmath>
+
+/**
+ * @brief Converts barometric pressure to a correction factor relative to standard sea-level pressure (760 mmHg).
+ * 
+ * @param pressure_mmhg Barometric pressure in mmHg (default: 760 mmHg).
+ * @return Correction factor to adjust DO saturation for pressure deviation.
+ * 
+ * Reference: 
+ * - https://water.usgs.gov/water-resources/memos/documents/WQ.2011.03.pdf
+ */
+double pressure_correction(double pressure_mmhg = 760.0) { return pressure_mmhg / 760.0; }
+
+/**
+ * @brief Computes the salinity correction factor for dissolved oxygen solubility.
+ * 
+ * This factor adjusts the oxygen solubility downward based on increased salinity.
+ * 
+ * @param temperature_k Water temperature in Kelvin.
+ * @param salinity_ppt Salinity in parts per thousand (ppt).
+ * @return Salinity correction factor (unitless).
+ * 
+ * Reference: Garcia & Gordon (1992), also adopted by USGS:
+ * - https://water.usgs.gov/water-resources/memos/documents/WQ.2011.03.pdf
+ */
+double salinity_correction(double temperature_k, double salinity_ppt)
+{
+    // base salinity correction
+    const double sal_corr_a = 0.017674;
+    // adjusts salinity correction based on temperature
+    const double sal_corr_b = 10.754;
+    // quadratic temp-based adjustment to salinity effect
+    const double sal_corr_c = 2140.7;
+
+    return std::exp(-salinity_ppt * (sal_corr_a - (sal_corr_b / temperature_k) +
+                                     (sal_corr_c / (temperature_k * temperature_k))));
+}
+
+/**
+ * @brief Calculates the saturation concentration of dissolved oxygen in freshwater at 1 atm and 0 ppt salinity.
+ * 
+ * This is based on the Benson & Krause (1984) equation recommended by USGS for high-accuracy DO solubility estimates.
+ * It returns the maximum possible DO (mg/L) at a given temperature in °C.
+ * 
+ * @param temperature_k Water temperature in degrees Kelvin.
+ * @return DO saturation in mg/L at standard atmospheric pressure and zero salinity.
+ * 
+ * Reference:
+ * - https://water.usgs.gov/water-resources/memos/documents/WQ.2011.03.pdf
+ */
+double calculate_do_saturation_fresh(double temperature_k)
+{
+    // Constants from Benson & Krause (1984), as recommended in USGS WQ.2011.03
```

Comment:
This function looks like it actually comes from the "Weiss Equations" section of the cited paper, rather than the "Benson & Krauss" section.  I'm not sure how different the two DO_0 equations are in practice.

Reply:
This would be equation (7) in https://water.usgs.gov/water-resources/memos/documents/WQ.2011.03.pdf